plugins/search-react/src/components/SearchFilter/SearchFilter.Autocomplete.tsx

```diff
@@ -79,6 +79,7 @@ export const AutocompleteFilter = (props: SearchAutocompleteFilterProps) => {
       }
       return { ...others };
     });
+    setPageCursor(undefined);
```

Comment:
To be sure I understood, this is for requesting the first results page whenever a filter change, right?

Reply:
If so, I believe doing this in the search context would benefit anyone who changes filters on other files, similar to this:
https://github.com/backstage/backstage/blob/535ca54f0e7ccc39eedefe084b94cb8aa42f7df9/plugins/search-react/src/context/SearchContext.tsx#L138-L144